src/main/scala/dpla/ingestion3/mappers/providers/WiMapping.scala

```diff
@@ -105,9 +107,17 @@ class WiMapping extends Mapping[NodeSeq] with XmlExtractor with IdMinter[NodeSeq
     }
   }
 
-  override def edmRights(data: Document[NodeSeq]): ZeroToOne[URI] =
+  override def edmRights(data: Document[NodeSeq]): ZeroToOne[URI] = {
     // Will quietly drop any invalid URIs
-    extractStrings(data \ "metadata" \\ "rights").find(r => Utils.isUri(r)).map(new URI(_))
+    val edmRights = (data \ "metadata" \\ "rights").map(rights => {
+      rights.prefix match {
+        case "edm" => rights.text
+        case _ => ""
+      }
+    })
+
+    edmRights.find(Utils.isUri).map(new URI(_))
```

Comment:
Sanity check - will the statement on line 119 drop empty strings?  I assume that empty strings count as invalid URIs so they will be dropped, correct?

Reply:
`find` returns the first value that meets the condition, `Utils.isUri` but I'm now noticing that the isUri method does not check for empty strings so this would result in an empty URI if the first value was an empty string. I'll fixup the `isUri` and this should be good.

Nice catch